pkg/utils/opts/armopts.go

```diff
@@ -35,21 +34,7 @@ func DefaultArmOpts() *arm.ClientOptions {
 
 func DefaultRetryOpts() policy.RetryOptions {
 	return policy.RetryOptions{
-		MaxRetries: 20,
-		// Note the default retry behavior is exponential backoff
-		RetryDelay: time.Second * 5,
-		// TODO: bsoghigian: Investigate if we want to leverage some of the status codes other than the defaults.
-		// the defaults are // StatusCodes specifies the HTTP status codes that indicate the operation should be retried.
-		// A nil slice will use the following values.
-		//   http.StatusRequestTimeout      408
-		//   http.StatusTooManyRequests     429
-		//   http.StatusInternalServerError 500
-		//   http.StatusBadGateway          502
-		//   http.StatusServiceUnavailable  503
-		//   http.StatusGatewayTimeout      504
-		// Specifying values will replace the default values.
-		// Specifying an empty slice will disable retries for HTTP status codes.
-		// StatusCodes: nil,
+		MaxRetries: -1,
```

Comment:
Consider leaving a comment here explaining what `-1` means, as there are APIs where this could easily mean "retry forever", but in the Go SDK it means no retries.

Reply:
yes, can update with comment. 

> Approved but we should make sure we have test coverage that demonstrates that the controller-runtime loop-based backoffs do the right thing here. Do we?

I would like to spend some time seperately investigating Performance and potentailly looking at controller options for retries separately on 429. Ideally karpenter core would stop sending requests for some timestamp when seeing 429, and wait to requeue until that interval passes, but need to think through this more. Will add when im at that stage. 

for now just removing the 20 retries we do for each failure case to lower throttling.

Reply:
Consider that the RP can return `retryAfter` to you to, which you can extract from the response headers and flow upstream to Karpenter (via `ctrl.Result{}` with a specific time), which in the case of throttling especially may be worth doing.

So while karpenter-core should probably be doing some basic backoff, etc, it might still be worth thinking about how our code can do the right thing here (or give Karpenter core the tools to do the right thing).